packages/react-components/src/components/resource-explorers/types/resource-explorer.ts

```diff
@@ -0,0 +1,96 @@
+import type {
+  ResourceTableSettings,
+  ResourceTableUserSettings,
+  TableResourceDefinition,
+} from './table';
+import {
+  PageSize,
+  PluralResourceName,
+  ResourceExplorerVariant,
+  ResourceName,
+  SelectionMode,
+  ShouldPersistUserCustomization,
+} from './common';
+import {
+  DropDownResourceDefinition,
+  ResourceDropDownSettings,
+} from './drop-down';
+
+/** Props common to all resource explorers. */
+export type CommonResourceExplorerProps<Resource> = {
+  /** TODO */
+  requestFns?: unknown;
```

Comment:
Is this type unknown because it's a commonly used interface across the more specific explorer interfaces?

Reply:
Pretty much. This is a little trick to ensure every resource explorer has a requestFns prop defined. The specific explorers override the value. It's unknown here because it's unusable internally without overriding.